site/gdocs/Chart.tsx

```diff
@@ -38,11 +39,26 @@ export default function Chart({
     const resolvedUrl = linkedChart.resolvedUrl
     const isExplorer = url.isExplorer
     const hasControls = url.queryParams.hideControls !== "true"
-    const height = d.height || (isExplorer && hasControls ? 700 : 575)
 
-    let config: GrapherProgrammaticInterface = {}
+    // applies to both charts and explorers
+    const common = {
+        // On mobile, we optimize for horizontal space by having Grapher bleed onto the edges horizontally.
+        // We want to do this for all stand-alone charts and charts in a Key Insights block, but not for charts
+        // listed in an All Charts block. The <Chart /> component is not used to render charts in an All Charts block,
+        // so we can just set this to true here.
+        shouldOptimizeForHorizontalSpace: true,
+    }
+
+    // props passed to explorers
+    const explorerProps: Pick<
+        ExplorerProps,
+        "shouldOptimizeForHorizontalSpace"
+    > = merge({}, common)
```

Comment:
When embedded in an article and viewed on mobile, we want to optimise for horizontal space (see screenshot in the [resp. PR](https://github.com/owid/owid-grapher/pull/2688)). In practise, when optimising for horizontal space, the horizontal frame padding is et to 0px and the left and right borders are removed, while the top and bottom borders stretch across the whole page. To this end, I have added a new property to Grapher called `shouldOptimizeForHorizontalSpace`.

Usually that works like this:
- The gdoc `<Chart />` component renders a figure with `data-grapher-src` set and the MultiEmbedder picks up the figure element and renders Grapher into it
- To pass information from the gdoc `<Chart />` component to Grapher, a stringified config is specified as `data-grapher-config` on the figure element, that data attribute is read by the MultiEmbedder, the parsed config is merged with other configs and then passed to Grapher when it's rendered. This is how, for example, the title or subtitle is passed to Grapher in narrative charts. And we use the same strategy to pass `shouldOptimizeForHorizontalSpace` (it's just another property).

That works for normal Grapher charts, but sometimes authors want to show one specific chart from an explorer. To do so, they link to a particular explorer view and hide the controls. Such charts look like ordinary Grapher charts but are in fact explorers. We want to optimize those charts for horizontal space as well. To do so, I have added a flag `shouldOptimizeForHorizontalSpace` to the explorer props.

Passing information from the gdocs `<Chart />` component to the explorer works exactly like passing information from `<Chart />` to Grapher:
- I added a new data attribute to the figure element, called `data-explorer-props`
- Whatever configuration is passed to `data-explorer-props` is picked up by the MultiEmbedder and then passed to the explorer
- For now, as a precaution, I only allow `shouldOptimizeForHorizontalSpace` to be passed
- An alternative would be to add `shouldOptimizeForHorizontalSpace` to the query string of an explorer but it would be odd, I think, to expose `shouldOptimizeForHorizontalSpace` like that